lib/color_spinor_pack.cu

```diff
@@ -84,133 +98,51 @@ namespace quda {
 	}
       }
       label[14] = '\0';
-      strcat(aux,label);
-    }
-
-    inline void apply(const qudaStream_t &stream) {
-      if (meta.Location() == QUDA_CPU_FIELD_LOCATION) {
-	if (arg.nDim == 5) GenericPackGhost<Float,block_float,Ns,Ms,Nc,Mc,5,Arg>(arg);
-	else GenericPackGhost<Float,block_float,Ns,Ms,Nc,Mc,4,Arg>(arg);
-      } else {
-	const TuneParam tp = tuneLaunch(*this, getTuning(), getVerbosity());
-	arg.nParity2dim_threads = arg.nParity*2*tp.aux.x;
-#ifdef JITIFY
-        using namespace jitify::reflection;
-        jitify_error = program->kernel("quda::GenericPackGhostKernel")
-          .instantiate(Type<Float>(),block_float,Ns,Ms,Nc,Mc,arg.nDim,(int)tp.aux.x,Type<Arg>())
-          .configure(tp.grid,tp.block,tp.shared_bytes,stream).launch(arg);
-#else
-        switch(tp.aux.x) {
-        case 1:
-	  if (arg.nDim == 5) qudaLaunchKernel(GenericPackGhostKernel<Float,block_float,Ns,Ms,Nc,Mc,5,1,Arg>, tp, stream, arg);
-	  else qudaLaunchKernel(GenericPackGhostKernel<Float,block_float,Ns,Ms,Nc,Mc,4,1,Arg>, tp, stream, arg);
-	  break;
-	case 2:
-	  if (arg.nDim == 5) qudaLaunchKernel(GenericPackGhostKernel<Float,block_float,Ns,Ms,Nc,Mc,5,2,Arg>, tp, stream, arg);
-	  else qudaLaunchKernel(GenericPackGhostKernel<Float,block_float,Ns,Ms,Nc,Mc,4,2,Arg>, tp, stream, arg);
-	  break;
-	case 4:
-	  if (arg.nDim == 5) qudaLaunchKernel(GenericPackGhostKernel<Float,block_float,Ns,Ms,Nc,Mc,5,4,Arg>, tp, stream, arg);
-	  else qudaLaunchKernel(GenericPackGhostKernel<Float,block_float,Ns,Ms,Nc,Mc,4,4,Arg>, tp, stream, arg);
-	  break;
-        }
-#endif
+      strcat(aux, label);
+      strcat(aux, ",spins_per_thread=");
+      u32toa(label, spins_per_thread(a));
+      strcat(aux, label);
+      strcat(aux, ",colors_per_thread=");
+      u32toa(label, colors_per_thread(a));
+      strcat(aux, label);
+
+      // compute number of number of work items we have to do
+      // unlike the dslash kernels, we include the fifth dimension here
+      for (int i = 0; i < 4; i++) {
+        if (!comm_dim_partitioned(i)) continue;
+        work_items += 2 * nFace * a.getDslashConstant().ghostFaceCB[i] * a.getDslashConstant().Ls; // 2 for forwards and backwards faces
```

Comment:
Did we ever reconcile the conversation about `work_items` vs `threads`? I feel like there are mixed conventions through this re-write, but iirc `threads` has been used the most.

Reply:
We did discuss this, and resolved it would be post GK.  In the context of this kernel, the name `work_items` is more appropriate since we can have multiple work items per thread.